processor/src/main/java/org/mapstruct/ap/internal/model/MethodReference.java

```diff
@@ -400,6 +397,9 @@ public boolean equals(Object obj) {
         if ( !Objects.equals( providingParameter, other.providingParameter ) ) {
             return false;
         }
+        if ( !Objects.equals( name, other.name ) ) {
+            return false;
+        }
```

Comment:
Not 100% sure if the name (and already been there: declaringMapper + providingParameter) is enough to differ between multiple method references. All unit tests are green.